### ISSUE TITLE
Intergate `form`

### ISSUE THREAD

**Issue by burrbull** [rust-embedded/svd2rust#424]
What are you think about integrate `form` functionality in `svd2rust` (optional)?

When this option is enabled we can do not merge mod TokenStreams in one (and then split with `form`) and write them in different files directly. I think this can give us significant speed-up in this case.

@therealprof

**Comment by therealprof** [comment]
Yes, I was thinking about that. I was also looking into integrating with `rustfmt` directly since the generated code is such a hot mess; quite obviously the code generation/parsing cabilities of syn/quote/proc_macro2 were never meant to generate code stored to disk but only non-persistent code to be compiled on-the-fly.
